lms/djangoapps/instructor/tests/views/test_instructor_dashboard.py

```diff
@@ -135,6 +137,31 @@ def has_instructor_tab(user, course):
         )
         assert has_instructor_tab(org_researcher, self.course)
 
+    @ddt.data(
+        ('staff', False, True),
+        ('staff', True, False),
+    )
+    @ddt.unpack
+    def test_discussion_tab(self, access_role, waffle_status, is_discussion_tab_available):
+        """
+        Verify that the Discussion tab only shows up when Pages & Resources flag is off for course
+        """
+        with override_waffle_flag(ENABLE_PAGES_AND_RESOURCES_MICROFRONTEND, waffle_status):
+            discussion_section = '<li class="nav-item"><button type="button" class="btn-link discussions_management" ' \
+                                 'data-section="discussions_management">Discussions</button></li>'
+
+            user = UserFactory.create(is_staff=True)
```

Comment:
I think `self.user` is available in the super class with staff status, have you checked that?

Reply:
If you are adding visibility in the test, than you can also use `AdminFactory` here.